Src/MakingSense.SafeBrowsing/CanonicalUrl.cs

```diff
@@ -0,0 +1,233 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace MakingSense.SafeBrowsing
+{
+    public class CanonicalUrl
```

Comment:
Nice class! I wonder if there is already a C# out of the box class with some of this functionality. Anyway, it looks very clear. Did you copy the code from another language implementation or do you have a link to the specification?